src/pkg/storage/s3.go

```diff
@@ -100,3 +121,107 @@ func (c S3Config) validate() error {
 
 	return nil
 }
+
+func s3Overrides(in map[string]string) map[string]string {
+	return map[string]string{
+		Bucket:                 in[Bucket],
+		Endpoint:               in[Endpoint],
+		Prefix:                 in[Prefix],
+		DoNotUseTLS:            in[DoNotUseTLS],
+		DoNotVerifyTLS:         in[DoNotVerifyTLS],
+		StorageProviderTypeKey: in[StorageProviderTypeKey],
+	}
+}
+
+func (c *S3Config) s3ConfigsFromStore(kvg KVStoreGetter) {
+	c.Bucket = cast.ToString(kvg.Get(BucketNameKey))
+	c.Endpoint = cast.ToString(kvg.Get(EndpointKey))
+	c.Prefix = cast.ToString(kvg.Get(PrefixKey))
+	c.DoNotUseTLS = cast.ToBool(kvg.Get(DisableTLSKey))
+	c.DoNotVerifyTLS = cast.ToBool(kvg.Get(DisableTLSVerificationKey))
+}
+
+func (c *S3Config) s3CredsFromStore(kvg KVStoreGetter) {
+	c.AccessKey = cast.ToString(kvg.Get(AccessKey))
+	c.SecretKey = cast.ToString(kvg.Get(SecretAccessKey))
+	c.SessionToken = cast.ToString(kvg.Get(SessionToken))
+}
+
+var _ Configurer = &S3Config{}
+
+func (c *S3Config) ApplyConfigOverrides(
+	kvg KVStoreGetter,
+	readConfigFromStore bool,
+	matchFromConfig bool,
+	overrides map[string]string,
+) error {
+	if readConfigFromStore {
+		c.s3ConfigsFromStore(kvg)
+
+		if b, ok := overrides[Bucket]; ok {
+			overrides[Bucket] = common.NormalizeBucket(b)
+		}
+
+		if p, ok := overrides[Prefix]; ok {
+			overrides[Prefix] = common.NormalizePrefix(p)
+		}
+
+		if matchFromConfig {
+			providerType := cast.ToString(kvg.Get(StorageProviderTypeKey))
+			if providerType != ProviderS3.String() {
+				return clues.New("unsupported storage provider: " + providerType)
+			}
+
+			if err := mustMatchConfig(kvg, s3constToTomlKeyMap, s3Overrides(overrides)); err != nil {
+				return clues.Wrap(err, "verifying s3 configs in corso config file")
+			}
+		}
+	}
+
+	c.s3CredsFromStore(kvg)
+
+	aws := credentials.AWS{
```

Comment:
Note: I've made some changes to this code here. [Original code](https://github.com/alcionai/corso/blob/edf753382ea99cb255f0ac1747adb97dcd5b8069/src/cli/config/storage.go#L99) for reference.

**Changes:**
1. Original code was not populating `S3Config.AWS` if the creds are sourced from env vars. Fixed it. Note that this wasn't causing any issues because kopia calls into minio later which patches[ empty creds from env vars](https://github.com/minio/minio-go/blob/5aaca534bbcfdfcecbfd2126c494b154fde16564/pkg/credentials/env_aws.go#L41).
This allows us to test for config correctness.

2. We were validating aws credentials. This is unnecessary since kopia/minio will do that for us later anyway. I've removed this.

Reply:
per 2/ can we get some examples of failed validation?  We want to ensure that the response to the user is able to be clearly understood as a credentials issue.